src/ploneintranet/workspace/browser/tiles/navigation.py

```diff
@@ -56,4 +53,48 @@ def render(self):
         return self.index()
 
     def __call__(self):
+        # TODO: use transient tiles with schema and data manager here
+        root = self.request.form.get('root')
+        if root:
+            self.root = api.content.get(UID=root)
+        else:
+            self.root = self.context
         return self.render()
+
+    @memoize
+    def _parent(self):
+        return aq_parent(self.root)
+
+    def current_node_is_root(self):
+        """
+        Is the current node the root of the workspace?
+        """
+        return IWorkspaceFolder.providedBy(self.root)
+
+    def parent_title(self):
+        return self._parent().Title()
+
+    def non_folderish_item_uids(self):
+        # TODO: Better way of determining folderish?
```

Comment:
It appears you should be able to use the `isPrincipiaFolderish` attribute/index:
http://stackoverflow.com/questions/15177286/how-to-add-the-is-folderish-attribute-to-dexterity-objects